### PR TITLE
Exporting images without backgrounds

### PR DESCRIPTION
**Pull request by csacsi**

### CODE REVIEW

**Comment by WiesnerPeti** [comment]
@csacsi I think, we could have used the gitflow branch naming here, so `feature/exportingImages`, what is your opinion here? I believe that is more descriptive :) (No need to rename the branch, this discussion is about future branches)

File: sketch2Code.sketchplugin [hunk 1]
```diff
@@ -103,9 +108,35 @@ function templateBinding(targetView)
...
+
+      [layer setIsVisible:true]
+
+      var layerName = [layer name]
+      var layerClass = NSStringFromClass([layer class])
+      var viewClass = mappedClassNameForLayer(layer)
+      var viewDesc = viewDescriptors[viewClass]
+      exportImageToFile(baseDir + "/" +  layerName, layer, viewDesc["export"]["format"])  
+
+      [layer setIsVisible:false]
```
Comment (WiesnerPeti) [inline sketch2Code.sketchplugin:128]: @csacsi Why is this needed? We set this above (line 113)

File: sketch2Code.sketchplugin [hunk 2]
```diff
@@ -79,6 +81,9 @@ function templateBinding(targetView)
     var inits = [NSMutableArray array]
     var addSubviews = [NSMutableArray array]
     var layouts = [NSMutableArray array]
+    layersToSave = [NSMutableArray array]
```
Comment (WiesnerPeti) [inline sketch2Code.sketchplugin]: @csacsi I think this should be at line 39, otherwise layersToSave won't be initialised if `templateBinding` is not called

File: sketch2Code.sketchplugin [hunk 3]
```diff
@@ -103,9 +108,35 @@ function templateBinding(targetView)
 
           var layoutString = layoutDefinition(layer, layerName)
           [layouts addObject:"\t" + layoutString]
+          [allLayers addObject:layer]
```
Comment (WiesnerPeti) [inline sketch2Code.sketchplugin]: @csacsi I think, this should be outside of the if condition (above line 93), because there could be layers, for that we do not have defined mapping in the descriptors, therefore they would not be added to allLayers, and won't be hidden

**Comment by WiesnerPeti** [comment]
@csacsi Somehow the template `arrow.pdf` won't be generated without background, we should have a look on this.

File: sketch2Code.sketchplugin (hunk 1 above)
Comment (csacsi) [inline sketch2Code.sketchplugin:128]: We just setting it to visible before the export and after it setting back to hidden, to prevent it from showing on other layers.
